app/controllers/line_items_controller.rb

```diff
@@ -0,0 +1,41 @@
+class LineItemsController < ApplicationController
+  before_action :get_cart, only: [:create]
+  before_action :get_line_item, only: [:destroy]
+
+  def create
+    @deal = Deal.find_by(id: params[:deal_id])
+    @line_item = @order.add_deal(params[:deal_id])
+
+    if @line_item.save
+      redirect_to deal_url(@deal), notice: I18n.t(:deal_added_to_cart, scope: [:flash, :notice], deal_title: @deal.title)
+    else
+      redirect_to deal_url(@deal), alert: @line_item.errors.full_messages.join("<br>")
```

Comment:
extract `@line_item.errors.full_messages.join("<br>")` in a method in line_item.rb
like `pretty_errors`